.github/workflows/ci-test.yml

```diff
@@ -45,6 +49,13 @@ jobs:
         if: matrix.os == 'windows-latest'
```

Comment:
I'd add another 'OS' in the matrix so that there is one windows run for powershell core and one for windows powershell so that it can finisher quicker.
What we've lost with this test approach is that the test job used to be its own job just consuming the binary as artefacts (and the repo checkout because of test files although they should've technically also been an artefact). This ensured that the built binaries were sufficient and weren't relying on something else left lying around from the build